activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -316,4 +316,20 @@ def set_defaults(target) # :doc:
   end
 end
 
-HashWithIndifferentAccess = ActiveSupport::HashWithIndifferentAccess
+class HashWithIndifferentAccess < ActiveSupport::HashWithIndifferentAccess
+  def initialize(*)
+    ActiveSupport::Deprecation.warn "HashWithIndifferentAccess is deprecated!" \
+                                    "Use ActiveSupport::HashWithIndifferentAccess instead."
+    super
+  end
+
+  def self.inherited(*)
+    ActiveSupport::Deprecation.warn "HashWithIndifferentAccess is deprecated!" \
+                                    "Use ActiveSupport::HashWithIndifferentAccess instead."
+    super
+  end
+
+  def encode_with(coder)
+    coder.represent_object(nil, ActiveSupport::HashWithIndifferentAccess.new(self))
```

Comment:
I think `Yaml.dump_tags[HashWithIndifferentAccess] = "!ruby/hash:ActiveSupport::HashWithIndifferentAccess"` would implement this.